src/Plugin.php

```diff
@@ -254,7 +254,7 @@ public function render( array $attributes, string $content, \WP_Block $block ):
 		$page             = empty( $_GET[ $page_key ] ) ? 1 : (int) $_GET[ $page_key ]; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
 		$page_parameter   = $inherit ? 'paged' : $page_key;
 		$block_query      = $inherit ? $wp_query : new \WP_Query( build_query_vars_from_query_block( $block, $page ) );
-		$max_pages        = '0' === $block->context['query']['pages'] ? $block_query->max_num_pages : (int) $block->context['query']['pages'];
+		$max_pages        = empty( $block->context['query']['pages'] ) ? $block_query->max_num_pages : (int) $block->context['query']['pages'];
```

Comment:
My initial fix was comparing `0 === ` as an integer, however I think `empty()` is better for this case since it accounts for both `0` and `"0"`, as well as other possible future scenarios like `NULL` or an empty string. 

So any falsy value ends up in `$block_query->max_num_pages` which is what got the items displayed.

Reply:
This looks good @ecairol